### PR TITLE
Further updates to the simulator docs

### PR DESCRIPTION
**Pull request by PeterJCLaw**
This includes a number of tidyups as well as content fixes. Suggest review by commit.

Fixes https://github.com/srobo/docs/issues/537.
Fixes https://github.com/srobo/competition-simulator/issues/407.

Builds on https://github.com/srobo/docs/pull/529.

### CODE REVIEW

**Comment by PeterJCLaw** [comment]
-> draft; apparently we want the simulator API & differences to be in the simulator section.

**Comment by PeterJCLaw** [comment]
Given the refactoring of the simulator section in the following PR, I think addressing the location of the API differences is better handled as a separate change -- likely building on that refactoring.

File: programming/arduino/custom_firmware.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ On this page we talk specifically about Arduinos.
 However this page is applicable to opening any device that shows up as a serial port.
 </div>
 
+<div class="info">
+  This documentation refers to a feature which is only available on the physical robot kits.
```
Comment (Adimote) [inline programming/arduino/custom_firmware.md:16]: Have you considered "This feature is not available on the robot simulator", for clarity? 

(This and all instances below)

**Review by Adimote — APPROVED** [review]

File: programming/arduino/custom_firmware.md (hunk 1 above)
Comment (PeterJCLaw) [reply programming/arduino/custom_firmware.md:16]: I hadn't; this was copy/pasted from the various other places we say this sort of thing. Happy to change, but suggest we do that separately for speed?

**Pull request merged**